torch/_prims_common/wrappers.py

```diff
@@ -191,8 +191,18 @@ def _out_wrapper(fn: Callable) -> Callable:
             )
         )
 
+        sig = inspect.signature(fn)
+        factory_kwargs = ("device", "dtype")
+        is_factory_fn = all(p in sig.parameters for p in factory_kwargs)
```

Comment:
I audited `native_functions.yaml` to check that this wouldn't fire inappropriately, and it appears to pass the audit, though I'm still a little nervous

Reply:
Thanks for checking - yeah I guess our assumption here is that any function that takes factory kwargs and accepts out arg should infer factory kwargs from out args if the factory kwargs are not specified. Improving our OpInfo (and sample input) coverage should help with checking this as well.